pkg/types/indicator.go

```diff
@@ -408,6 +408,41 @@ func Array(a Series, limit ...int) (result []float64) {
 	return
 }
 
+type RegressionType string
+
+const (
+	RegressionTypeConstant              RegressionType = "c"
+	RegressionTypeConstantTrend         RegressionType = "ct"
+	RegressionTypeConstantTrendSeasonal RegressionType = "ctt"
+	RegressionTypeNone                  RegressionType = "n"
+)
+
+// Ordinary Least Squares fit result, only support 1d array
+func OSL(a SeriesExtend, b SeriesExtend, n int) (float64, float64) {
+	if a.Length() < n {
```

Comment:
How about setting `n` to `a.Length()` if `a.Length() < n`?
So that you can continue the calculation for a result still.

Reply:
you're correct. The other implementations are following the same rule.
done.